src/simulators/statevector/qv_avx2.cpp

```diff
@@ -1172,16 +1166,16 @@ Avx apply_diagonal_matrix_avx<double>(double* qv_data_,
   auto qv_data = _to_complex(qv_data_);
   const auto input_vec = _to_complex(vec_);
 
-#pragma omp parallel if (omp_threads > 1) num_threads(omp_threads)
-  {
+  // conservatively allocate temporal thread-local memory
+  // some omp implementation does not correctly manage threadprivate
+  auto omp_max_threads = omp_get_max_threads();
```

Comment:
Can you absolutely rely on OpenMP being available at build time?  The `#include <omp.h>` at the top is protected in an `#ifdef _OPENMP`.  The previous `#pragma omp` forms silently degrade to no-ops if OpenMP isn't there, but this would be a compilation error due to the missing `omp_get_max_threads` symbol.

Reply:
Thanks. Very good point.